source-code/core/app/src/wrappers/withSolidReactivity.test.ts

```diff
@@ -0,0 +1,337 @@
+import { describe, it, expect } from "vitest"
+import { createInlang } from "../createInlang.js"
+import { createMockNodeishFs } from "@inlang/plugin/test"
+import type { InlangConfig } from "@inlang/config"
+import type { Message, Plugin } from "@inlang/plugin"
+import type { ImportFunction, InlangModule } from "@inlang/module"
+import { createRoot, createEffect, from } from "../solid.js"
+import { withSolidReactivity } from "./withSolidReactivity.js"
+import type { LintRule } from "@inlang/lint"
+
+describe("config", () => {
+	it("should react to changes to config", async () => {
+		const fs = await createMockNodeishFs()
+		await fs.writeFile("./inlang.config.json", JSON.stringify(config))
+		await createRoot(async () => {
+			const inlang = withSolidReactivity(
+				await createInlang({
+					configPath: "./inlang.config.json",
+					nodeishFs: fs,
+					_import: $import,
+				}),
+				{ from },
+			)
+			let counter = 0
+
+			createEffect(() => {
+				// 2 times because effect creation + set
+				inlang.config()
+				counter += 1
+			})
+
+			inlang.setConfig({ ...inlang.config(), languageTags: ["en", "de"] })
+			expect(counter).toBe(2)
+		})
+	})
+})
+
+describe("meta", () => {
+	it("should react to changes to config", async () => {
+		await createRoot(async () => {
+			const fs = await createMockNodeishFs()
+			await fs.writeFile("./inlang.config.json", JSON.stringify(config))
+			const inlang = withSolidReactivity(
+				await createInlang({
+					configPath: "./inlang.config.json",
+					nodeishFs: fs,
+					_import: $import,
+				}),
+				{ from },
+			)
+			let counterPlugins = 0
+			let counterLint = 0
+
+			createEffect(() => {
+				// 2 times because effect creation + set
+				inlang.meta.plugins()
+				counterPlugins += 1
+			})
+
+			createEffect(() => {
+				// 2 times because effect creation + set
+				inlang.meta.lintRules()
+				counterLint += 1
+			})
+
+			inlang.setConfig({ ...inlang.config(), languageTags: ["en", "fr"] })
+			setTimeout(() => {
```

Comment:
@NilsJacobsen FYI this was never awaited and therefore the tests didn't check anything.